sql/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -125,37 +128,46 @@ public ExecutionPlan build(PlannerContext plannerContext,
             rightExecutionPlan = tmp;
         }
 
+        Tuple<List<Symbol>, List<Symbol>> hashInputs =
+            extractHashJoinInputsFromJoinSymbolsAndSplitPerSide(tablesSwitched);
+
         ResultDescription leftResultDesc = leftExecutionPlan.resultDescription();
-        ResultDescription rightResultDesc = rightExecutionPlan.resultDescription();
-        Collection<String> nlExecutionNodes = ImmutableSet.of(plannerContext.handlerNode());
+        Collection<String> joinExecutionNodes;
+        boolean isDistributed = true;
+        if (isDistributed) {
```

Comment:
isDistributed is currently always true, so the non-distributed branches could be removed. I'm not sure if we should leave them or remove them.

Reply:
I'd remove it, we've git to take a look at the non-distributed version if we want to.